### ISSUE TITLE
Error: The requested module does not provide an export named 'default'

### ISSUE THREAD

**Issue by jcmonnin** [GIShub4/solid-map-gl#6]
When setting up a simple test project, I get following error:
![image](https://user-images.githubusercontent.com/1336324/169485699-e3a69228-d8f8-493f-8528-61ac7a06d2a1.png)

Steps to repoduce:

- `npx degit solidjs/templates/ts my-app`
- `npm install --save solid-map-gl`
- Paste any of the examples of the README to App.tsx
- `npm start`

It looks like a bundler issue, however I'm not very familiar with that. Might be related to this:
https://github.com/mapbox/mapbox-gl-js/issues/8676

**Comment by KaiHuebner** [comment]
Hi Jean-Claude,

After a bit of research I found a solution.
I can replicate your issue when I'm using `npm` or `yarn`.

For bundling `solid-map-gl` I'm using [`pnpm`](https://pnpm.io) and [`rollup`](https://rollupjs.org/) which seem to have no problem with the Mapbox issue.

As a workaround until Mapbox fixes the issue please use `pnpm`.

```
- npx degit solidjs/templates/ts my-app
- pnpm i
- pnpm add solid-map-gl
- pnpm dev
```

**Comment by jcmonnin** [comment]
Hi Kay,

Thanks for looking into this. Unfortunately, for me, pnpm didn't make a difference.  I created a fresh test project as you described and i I still get the error. My initial project where I wanted to use `solid-map-gl` uses pnpm.

I could find following workaround:

- Add directly `mapbox-gl` as project dependency (eg. `pnpm add mabox-gl`)
- Add following lines to `vite.config.ts`:
````
  optimizeDeps: {
    include: ['mapbox-gl']
  }
````

I got the hint for using `optimizeDeps` from here: https://github.com/vitejs/vite/issues/813.
